lib/rx/util/health_check_authorization.rb

```diff
@@ -0,0 +1,35 @@
+module Rx
+  module Util
+    class HealthCheckAuthorization
+      def initialize(env, authorization)
+        @authorization = authorization
+        @env = env
+      end
+
+      def ok?
+        return custom_authorization(@env, @authorization) if @authorization.is_a?(Proc)

```

Comment:
this is purely stylistic, so feel free to tell me that I'm wrong but what do you think of using a `case` statement here instead of having different methods? e.g.

```ruby
case @authorization
  when NilClass
    true
  when Proc
    @authorization.call(@env)
  when String
    @authorization == @env["HTTP_AUTHORIZATION"]
  else
    raise StandardError.new("...")
end
```

I feel like that does a little better job of self-documenting which types are accepted and what the behaviors of each are.

Reply:
It didn't come to my mind to write it this way but I like a better in this way. changed now